src/v/config/convert.h

```diff
@@ -314,4 +316,32 @@ struct convert<model::partition_autobalancing_mode> {
     }
 };
 
+template<typename T>
+struct convert<std::unordered_map<typename T::key_type, T>> {
```

Comment:
Maybe we could use a concept to ensure at least T contains a type definition named `key_type`. At most we can ensure it contains the min necessary properties to be a valid key type in an `std::unordered_map`, like being equality comparable, etc